### PR TITLE
nodejs append records api

### PR DESCRIPTION
**Pull request by gsilvestrin**

### CODE REVIEW

File: node/package.json [hunk 1]
```diff
@@ -6,7 +6,7 @@
   "types": "dist/index.d.ts",
   "scripts": {
     "tsc": "tsc -b",
-    "build": "cargo-cp-artifact --artifact cdylib vectordb-node index.node -- cargo build --message-format=json-render-diagnostics",
+    "build": "cargo-cp-artifact --artifact cdylib vectordb-node darwin_arm64.node -- cargo build --message-format=json-render-diagnostics",
```
[the remark this answers is not in the document]
Comment (eddyxu) [reply node/package.json:9]: it does not work on linux/x86?

File: rust/vectordb/src/table.rs [hunk 2]
```diff
@@ -42,32 +43,65 @@ impl Table {
...
+    ///
+    /// * `batches` RecordBatch to be saved in the Table
+    /// * `write_mode` Append / Overwrite existing records. Default: Append
+    /// # Returns
+    ///
+    /// * The number of rows added
+    pub async fn add(
+        &mut self,
+        mut batches: Box<dyn RecordBatchReader>,
+        write_mode: Option<WriteMode>
```
[the remark this answers is not in the document]
Comment (changhiskhan) [reply rust/vectordb/src/table.rs:98]: yeah, or even possibly have separate `add` and `overwrite` methods (for both python and js)? can be refactored later though.

[the remark this answers is not in the document]
Comment (changhiskhan) [reply rust/vectordb/src/table.rs:98]: clarification: `create` mode is probably unnecessary here.

**Review by eddyxu — APPROVED** [review]

**Comment by martinnormark** [comment]
Awesome to see this! Are you planning to cut a release to npm anytime soon?

**Pull request merged**